src/btree/bt_vrfy.c

```diff
@@ -375,18 +375,22 @@ __verify_tree(WT_SESSION_IMPL *session, WT_REF *ref, WT_CELL_UNPACK *addr_unpack
     __wt_verbose(session, WT_VERB_VERIFY, "%s %s", __wt_page_addr_string(session, ref, vs->tmp1),
       __wt_page_type_string(page->type));
 
-    /* Optionally dump the address/timestamp information. */
-    if (vs->dump_address || vs->dump_time_pairs) {
+    /*
+     * Optionally dump the address/time pair information. dump_blocks prints time pairs and
+     * dump_pages dumps addresses and time pairs so here we try to avoid printing duplicate
+     * information when multiple -d options are provided.
+     */
+    if ((vs->dump_address || (vs->dump_time_pairs && !vs->dump_blocks)) && !vs->dump_pages) {
         WT_RET(__wt_msg(session, "%s", __wt_page_type_string(page->type)));
         if (vs->dump_address)
             WT_RET(__wt_msg(session, "\t%s", __wt_page_addr_string(session, ref, vs->tmp1)));
-        if (vs->dump_time_pairs) {
+        if (vs->dump_time_pairs && !vs->dump_blocks) {
             addr = ref->addr;
             if (addr != NULL)
-                WT_RET(__wt_msg(session, "\t%s/%" PRIu64 ",%s/%" PRIu64,
-                  __wt_timestamp_to_string(addr->oldest_start_ts, ts_string),
-                  addr->oldest_start_txn, __wt_timestamp_to_string(addr->newest_stop_ts, ts_string),
-                  addr->newest_stop_txn));
+                WT_RET(__wt_msg(session, "\t%s,%s", __wt_time_pair_to_string(addr->oldest_start_ts,
+                                                      addr->oldest_start_txn, tp_string[0]),
+                  __wt_time_pair_to_string(
+                                  addr->newest_stop_ts, addr->newest_stop_txn, tp_string[1])));
```

Comment:
I'd personally be inclined to disallow multiple `-d debug` options, it doesn't make much sense to me and (as noted), the output will be corrupted, and this only fixes one case. I won't fight about it, though.

Reply:
removed in https://github.com/wiredtiger/wiredtiger/pull/5138